main/provider/index.ts

```diff
@@ -448,33 +450,36 @@ export class Provider extends EventEmitter {
       const gas = gasFees(rawTx)
       const chainConfig = this.connection.connections['ethereum'][parseInt(rawTx.chainId)].chainConfig
 
-      const estimateGas = rawTx.gasLimit
-        ? Promise.resolve(rawTx)
-        : this.getGasEstimate(rawTx)
-            .then((gasLimit) => ({ ...rawTx, gasLimit }))
-            .catch((err) => {
-              approvals.push({
-                type: ApprovalType.GasLimitApproval,
-                data: {
-                  message: err.message,
-                  gasLimit: '0x00'
-                }
-              })
-
-              return { ...rawTx, gasLimit: '0x00' }
+      const estimateGasLimit = async () => {
+        if (rawTx.gasLimit) return rawTx.gasLimit
+        try {
+          return await this.getGasEstimate(rawTx)
+        } catch (error) {
+          if (error instanceof Error)
+            approvals.push({
+              type: ApprovalType.GasLimitApproval,
+              data: {
+                message: error.message,
+                gasLimit: '0x00'
+              }
             })
+          return '0x00'
+        }
+      }
 
-      estimateGas
-        .then((tx) => {
-          const populatedTransaction = populateTransaction(tx, chainConfig, gas)
+      const [gasLimit, recipientType] = await Promise.all([
+        estimateGasLimit(),
```

Comment:
nice cleanup here. we can also use a coalescing operator and remove the check in the function above. might be able to do it succinctly like:
`const gasLimit = rawTx.gasLimit ?? await estimateGasLimit()`

Reply:
or maybe the `||` operator if gas limit can be an empty string, not sure what the potential values are